File: apps/ios/ConstructionManager/ConstructionManager/Theme/AppColors.swift
//
//  AppColors.swift
//  ConstructionManager
//
//  Design system colors matching web app with dark mode support
//

import SwiftUI

struct AppColors {
    // MARK: - Primary Blue
    static let primary50 = Color(hex: "eff6ff")
    static let primary100 = Color(hex: "dbeafe")
    static let primary200 = Color(hex: "bfdbfe")
    static let primary300 = Color(hex: "93c5fd")
    static let primary400 = Color(hex: "60a5fa")
    static let primary500 = Color(hex: "3b82f6")
    static let primary600 = Color(hex: "2563eb")
    static let primary700 = Color(hex: "1d4ed8")
    static let primary800 = Color(hex: "1e40af")
    static let primary900 = Color(hex: "1e3a8a")

    // MARK: - Gray Scale (Adaptive for dark mode)
    static let gray50 = Color(light: Color(hex: "f9fafb"), dark: Color(hex: "1f2937"))
    static let gray100 = Color(light: Color(hex: "f3f4f6"), dark: Color(hex: "374151"))
    static let gray200 = Color(light: Color(hex: "e5e7eb"), dark: Color(hex: "4b5563"))
    static let gray300 = Color(light: Color(hex: "d1d5db"), dark: Color(hex: "6b7280"))
    static let gray400 = Color(hex: "9ca3af")
    static let gray500 = Color(hex: "6b7280")
    static let gray600 = Color(hex: "4b5563")
    static let gray700 = Color(hex: "374151")
    static let gray800 = Color(hex: "1f2937")
    static let gray900 = Color(hex: "111827")

    // MARK: - Semantic Colors
    static let success = Color(hex: "22c55e")
    static let successLight = Color(hex: "dcfce7")
    static let warning = Color(hex: "eab308")
    static let warningLight = Color(hex: "fef9c3")
    static let error = Color(hex: "ef4444")
    static let errorLight = Color(hex: "fee2e2")
    static let info = Color(hex: "3b82f6")
    static let infoLight = Color(hex: "dbeafe")

    // MARK: - Construction Specific
    static let orange = Color(hex: "f97316")
    static let orangeLight = Color(hex: "fff7ed")

    // MARK: - Purple (for team/people features)
    static let purple = Color(hex: "8b5cf6")
    static let purpleLight = Color(hex: "ede9fe")

    // MARK: - Adaptive Background Colors (supports dark mode via Color.init with light/dark)
    static let background = Color(light: Color(hex: "f9fafb"), dark: Color(hex: "111827"))
    static let cardBackground = Color(light: .white, dark: Color(hex: "1f2937"))
    static let sidebarBackground = Color(hex: "111827")

    // MARK: - Adaptive Text Colors (supports dark mode)
    static let textPrimary = Color(light: Color(hex: "111827"), dark: Color(hex: "f9fafb"))
    static let textSecondary = Color(light: Color(hex: "4b5563"), dark: Color(hex: "9ca3af"))
    static let textTertiary = Color(light: Color(hex: "6b7280"), dark: Color(hex: "6b7280"))
    static let textMuted = Color(light: Color(hex: "9ca3af"), dark: Color(hex: "4b5563"))
}

// MARK: - Color Extension for Hex Support
extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (1, 1, 1, 0)
        }
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }

    /// Creates an adaptive color that changes based on light/dark mode
    init(light: Color, dark: Color) {
        self.init(uiColor: UIColor { traitCollection in
            switch traitCollection.userInterfaceStyle {
            case .dark:
                return UIColor(dark)
            default:
                return UIColor(light)
            }
        })
    }
}

File: apps/android/construction-android/app/src/main/java/com/constructionpro/app/ui/theme/AppColors.kt
package com.constructionpro.app.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.ReadOnlyComposable
import androidx.compose.ui.graphics.Color

/**
 * Semantic color system matching iOS design.
 * Provides adaptive colors that change based on light/dark theme.
 *
 * Usage:
 *   AppColors.textPrimary   // Adaptive text color
 *   AppColors.cardBackground // Adaptive card background
 */
object AppColors {
    // MARK: - Primary Blue (same as Color.kt, re-exported for consistency)
    val primary50 = Primary50
    val primary100 = Primary100
    val primary200 = Primary200
    val primary300 = Primary300
    val primary400 = Primary400
    val primary500 = Primary500
    val primary600 = Primary600
    val primary700 = Primary700
    val primary800 = Primary800
    val primary900 = Primary900

    // MARK: - Semantic Colors
    val success = Color(0xFF22C55E)
    val successLight = Color(0xFFDCFCE7)
    val warning = Color(0xFFEAB308)
    val warningLight = Color(0xFFFEF9C3)
    val error = Color(0xFFEF4444)
    val errorLight = Color(0xFFFEE2E2)
    val info = Color(0xFF3B82F6)
    val infoLight = Color(0xFFDBEAFE)

    // MARK: - Construction Specific
    val orange = Color(0xFFF97316)
    val orangeLight = Color(0xFFFFF7ED)

    // MARK: - Purple (for team/people features)
    val purple = Color(0xFF8B5CF6)
    val purpleLight = Color(0xFFEDE9FE)

    // MARK: - Adaptive Colors (Composable)

    /** Primary text color - adapts to theme */
    val textPrimary: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFFF9FAFB) else Color(0xFF111827)

    /** Secondary text color - adapts to theme */
    val textSecondary: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF9CA3AF) else Color(0xFF4B5563)

    /** Tertiary/muted text color */
    val textTertiary: Color
        @Composable
        @ReadOnlyComposable
        get() = Color(0xFF6B7280)

    /** Muted text color - adapts to theme */
    val textMuted: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF4B5563) else Color(0xFF9CA3AF)

    /** Screen background color - adapts to theme */
    val background: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF111827) else Color(0xFFF9FAFB)

    /** Card/surface background - adapts to theme */
    val cardBackground: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF1F2937) else Color.White

    /** Surface variant for subtle differentiation - adapts to theme */
    val surfaceVariant: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF374151) else Color(0xFFF3F4F6)

    /** Sidebar/navigation background */
    val sidebarBackground = Color(0xFF111827)

    // MARK: - Gray Scale (adaptive)

    val gray50: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF1F2937) else Color(0xFFF9FAFB)

    val gray100: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF374151) else Color(0xFFF3F4F6)

    val gray200: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF4B5563) else Color(0xFFE5E7EB)

    val gray300: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF6B7280) else Color(0xFFD1D5DB)

    // Non-adaptive grays
    val gray400 = Color(0xFF9CA3AF)
    val gray500 = Color(0xFF6B7280)
    val gray600 = Color(0xFF4B5563)
    val gray700 = Color(0xFF374151)
    val gray800 = Color(0xFF1F2937)
    val gray900 = Color(0xFF111827)

    // MARK: - Divider/Border Colors
    val divider: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF374151) else Color(0xFFE5E7EB)

    val border: Color
        @Composable
        @ReadOnlyComposable
        get() = if (isSystemInDarkTheme()) Color(0xFF4B5563) else Color(0xFFD1D5DB)
}

/**
 * Status colors for daily logs, time entries, etc.
 */
object StatusColors {
    val draft = Color(0xFF6B7280)       // Gray
    val submitted = Color(0xFF3B82F6)   // Blue
    val approved = Color(0xFF22C55E)    // Green
    val rejected = Color(0xFFEF4444)    // Red
    val pending = Color(0xFFF59E0B)     // Amber
    val active = Color(0xFF22C55E)      // Green
    val onHold = Color(0xFFF59E0B)      // Amber
    val completed = Color(0xFF3B82F6)   // Blue
    val cancelled = Color(0xFF6B7280)   // Gray

    fun forStatus(status: String): Color = when (status.uppercase()) {
        "DRAFT" -> draft
        "SUBMITTED" -> submitted
        "APPROVED" -> approved
        "REJECTED" -> rejected
        "PENDING" -> pending
        "ACTIVE" -> active
        "ON_HOLD", "ONHOLD" -> onHold
        "COMPLETED" -> completed
        "CANCELLED", "CANCELED" -> cancelled
        else -> draft
    }

    fun backgroundForStatus(status: String): Color = when (status.uppercase()) {
        "DRAFT" -> Color(0xFFF3F4F6)
        "SUBMITTED" -> Color(0xFFDBEAFE)
        "APPROVED" -> Color(0xFFDCFCE7)
        "REJECTED" -> Color(0xFFFEE2E2)
        "PENDING" -> Color(0xFFFEF3C7)
        "ACTIVE" -> Color(0xFFDCFCE7)
        "ON_HOLD", "ONHOLD" -> Color(0xFFFEF3C7)
        "COMPLETED" -> Color(0xFFDBEAFE)
        "CANCELLED", "CANCELED" -> Color(0xFFF3F4F6)
        else -> Color(0xFFF3F4F6)
    }
}
